service/frontend/workflowHandler.go

```diff
@@ -2747,10 +2835,15 @@ func (wh *WorkflowHandler) SignalWithStartWorkflowExecution(
 		return nil, wh.error(err, scope, tags...)
 	}
 
+	isolationGroup := wh.getIsolationGroup(ctx, domainName)
+	if !wh.isIsolationGroupHealthy(ctx, domainName, isolationGroup) {
+		return nil, wh.error(&types.BadRequestError{fmt.Sprintf("Domain %s is drained from isolation group %s.", domainName, isolationGroup)}, scope, tags...)
+	}
+
 	resp, err = wh.GetHistoryClient().SignalWithStartWorkflowExecution(ctx, &types.HistorySignalWithStartWorkflowExecutionRequest{
 		DomainUUID:             domainID,
 		SignalWithStartRequest: signalWithStartRequest,
-		PartitionConfig:        partition.ConfigFromContext(ctx),
+		PartitionConfig:        wh.getPartitionConfig(ctx, domainName),
```

Comment:
This I'm a little confused by... shouldn't it be their current isolation group, where they going to be routed to?